airflow/models/dagrun.py

```diff
@@ -839,7 +841,7 @@ def verify_integrity(
         from airflow.settings import task_instance_mutation_hook
 
         # Set for the empty default in airflow.settings -- if it's not set this means it has been changed
-        hook_is_noop = getattr(task_instance_mutation_hook, 'is_noop', False)
+        hook_is_noop: Literal[True, False] = getattr(task_instance_mutation_hook, 'is_noop', False)
```

Comment:
I believe `bool` would work, Mypy should be smart enough to automatically coerce.

Reply:
I tried, it didn't unfortunately.

Reply:
Changing it back returns:
```
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
airflow/models/dagrun.py:867: error: No overload variant of "_get_task_creator"
of "DagRun" matches argument types "Dict[str, int]", "Callable[[Any], Any]",
"bool"  [call-overload]
            task_creator = self._get_task_creator(created_counts, task_ins...
                           ^
airflow/models/dagrun.py:867: note: Possible overload variants:
airflow/models/dagrun.py:867: note:     def _get_task_creator(self, created_counts: Dict[str, int], ti_mutation_hook: Callable[..., Any], hook_is_noop: Literal[True]) -> Callable[[Union[BaseOperator, MappedOperator], Tuple[int, ...]], Iterator[Dict[str, Any]]]
airflow/models/dagrun.py:867: note:     def _get_task_creator(self, created_counts: Dict[str, int], ti_mutation_hook: Callable[..., Any], hook_is_noop: Literal[False]) -> Callable[[Union[BaseOperator, MappedOperator], Tuple[int, ...]], Iterator[TaskInstance]]
Found 1 error in 1 file (checked 1 source file)
````

Reply:
Added a note to be more explicit why it has to be `Literal[True, False]`